src/Testcontainers/Builders/TestcontainersBuilderAzuriteExtension.cs

```diff
@@ -0,0 +1,146 @@
+namespace DotNet.Testcontainers.Builders
+{
+  using System.Collections.Generic;
+  using System.IO;
+  using DotNet.Testcontainers.Configurations;
+  using DotNet.Testcontainers.Containers;
+  using JetBrains.Annotations;
+
+  /// <summary>
+  /// This class applies the extended Testcontainer configurations for Azurite.
+  /// </summary>
+  [PublicAPI]
+  public static class TestcontainersBuilderAzuriteExtension
+  {
+    public static ITestcontainersBuilder<AzuriteTestcontainer> WithAzurite(this ITestcontainersBuilder<AzuriteTestcontainer> builder, AzuriteTestcontainerConfiguration configuration)
+    {
+      builder = builder
+        .WithImage(configuration.Image)
+        .WithWaitStrategy(configuration.WaitStrategy)
+        .ConfigureContainer(container =>
+        {
+          container.ContainerBlobPort = configuration.BlobServiceOnlyEnabled || configuration.AllServicesEnabled ? configuration.BlobContainerPort : 0;
+          container.ContainerQueuePort = configuration.QueueServiceOnlyEnabled || configuration.AllServicesEnabled ? configuration.QueueContainerPort : 0;
+          container.ContainerTablePort = configuration.TableServiceOnlyEnabled || configuration.AllServicesEnabled ? configuration.TableContainerPort : 0;
+        });
+
+      if (configuration.BlobServiceOnlyEnabled || configuration.AllServicesEnabled)
+      {
+        builder = builder
+          .WithExposedPort(configuration.BlobContainerPort)
+          .WithPortBinding(configuration.BlobPort, configuration.BlobContainerPort);
+      }
+
+      if (configuration.QueueServiceOnlyEnabled || configuration.AllServicesEnabled)
+      {
+        builder = builder
+          .WithExposedPort(configuration.QueueContainerPort)
+          .WithPortBinding(configuration.QueuePort, configuration.QueueContainerPort);
+      }
+
+      if (configuration.TableServiceOnlyEnabled || configuration.AllServicesEnabled)
+      {
+        builder = builder
+          .WithExposedPort(configuration.TableContainerPort)
+          .WithPortBinding(configuration.TablePort, configuration.TableContainerPort);
+      }
+
+      if (!string.IsNullOrWhiteSpace(configuration.Location))
```

Comment:
I am not sure if I follow you. This condition checks if user wants bind custom location. In case he provides incorrect folder path then I expeect that it will fail later somewhere. But I should set that folder because user expects it. Should I validate path before I will set bind?

Reply:
Fixed in [31aaf9c](https://github.com/testcontainers/testcontainers-dotnet/pull/555/commits/31aaf9ca336fbf76c035468a3e64378c9087ccbb)
Please check if I understand correctly to your comment.